src/cellfft/source/cellfft_get_kernel.cpp

```diff
@@ -0,0 +1,488 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2019 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+// clang-format off
+#include <sstream>
+#include <miopen/kernel_info.hpp>
+#include <miopen/gcn_asm_utils.hpp>
+#include "../include/cellfft_get_kernel.hpp"
+
+#define START_R2C_S   0
+#define START_C2R_S   (START_R2C_S+8)
+#define START_R2C_X   (START_C2R_S+22)
+#define START_R2C_Xg  (START_R2C_X+6)
+#define START_C2R_X   (START_R2C_Xg+8)
+#define START_R2C     (START_C2R_X+4)
+#define START_R2C_PAD (START_R2C+96)
+#define START_C2R     (START_R2C_PAD+48)
+
+static const char* g_knames[]=
+{
+    "sfft4x4_r2c_perm_s3x3",
```

Comment:
AFAICS we have hundreds of kernels in one .s file. After assembly, we have a large code object, but only a few kernels will be used. I'm afraid we are wasting time for loading the excessively big module. Is it possible to split the asm source file? Have you considered how this would affect performance?

Reply:
Cellfft is the most matching combination from a large number of kernels. It is difficult to group and split separately, which will make the problem very complicated. The size binary object (if include gfx908) is about 1.6M,so I think it will only slightly affect the initialization time

Reply:
> It is difficult to group and split separately, which will make the problem very complicated.

Hmm... I do not see any need to _group_ kernels. One source file per kernel should be fine. If you have some kernel generation tool, that shouldn't be a problem, right?

> The size binary object (if include gfx908) is about 1.6M,so I think it will only slightly affect the initialization time

I tend to agree with you. However I would prefer to really check this. I do not know if ROCm is good with big files containing many small kernels.

Reply:
> However I would prefer to really check this. I do not know if ROCm is good with big files containing many small kernels.

Any results of the check?